src/s3path.jl

```diff
@@ -102,7 +102,7 @@ function Base.join(prefix::S3Path, pieces::AbstractString...)
 
     return S3Path(
         tuple(segments...),
-        prefix.root,
+        "/",
```

Comment:
explain this change so I understand?

Reply:
I'm not 100% here. but the thought is this:

if we start with `foo`:
```julia
julia> foo = p"s3://foo"
p"s3://foo"

julia> foo.root
""
```
and we `new_foobar = join(foo, "bar")` correctly, so as to get it `== foobar`:
```julia
julia> foobar = p"s3://foo/bar"
p"s3://foo/bar"

julia> foobar.root
"/"
```
then we have to set the root of `new_foobar` to `"/"`

Reply:
i.e. without this change (the thing this PR is fixing):
```julia
julia> new_foobar = join(foo, "bar")
p"s3://foobar"

julia> new_foobar.root
""

julia> awkward_foobar = join(foo * "/", "bar")
p"s3://foo/bar"

julia> awkward_foobar.root
"/"
```